### PR TITLE
use enclose organization for source registry

### PR DESCRIPTION
**Pull request by rcerven**
Signed-off-by: Robert Cerven <rcerven@redhat.com>

### CODE REVIEW

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 1]
```diff
@@ -80,6 +81,10 @@ def run(self):
 
             image = self._ensure_image_registry(image)
 
+            organization = get_registries_organization(self.workflow)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_pull_base_image.py]: Let's do this outside the for-loop.

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 2]
```diff
@@ -80,6 +81,10 @@ def run(self):
...
+            if organization:
+                image.enclose(organization)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_pull_base_image.py]: This will change the ImageName object in the keys of `self.workflow.builder.parent_images`. I think that's what we want.

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 1 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_pull_base_image.py]: right

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 2 above)
Comment (twaugh) [reply atomic_reactor/plugins/pre_pull_base_image.py]: You *do* think that's what we want, or you don't?

File: atomic_reactor/schemas/config.json [hunk 3]
```diff
@@ -449,7 +449,6 @@
                 "description": "Registry url",
                 "type": "string"
             },
-            "organization": {"$ref": "#/definitions/organization"},
```
Comment (twaugh) [inline atomic_reactor/schemas/config.json]: What's the reason for this change?

Comment (lcarva) [reply atomic_reactor/schemas/config.json]: organization now is set via the `registries_organization` setting.

Comment (rcerven) [reply atomic_reactor/schemas/config.json]: we are using genera 'registries_organization'

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 2 above)
Comment (twaugh) [reply atomic_reactor/plugins/pre_pull_base_image.py]: I agree, the second one is better.

File: atomic_reactor/plugins/pre_pull_base_image.py [hunk 4]
```diff
@@ -91,9 +98,12 @@ def run(self):
                 new_image = image
             else:
                 new_image = self._pull_and_tag_image(image, build_json, str(nonce))
+            self.workflow.builder.recreate_parent_images()
             self.workflow.builder.parent_images[parent] = new_image
 
             if is_base_image:
+                if organization:
+                    self.workflow.builder.original_base_image.enclose(organization)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_pull_base_image.py]: Let's add a comment explaining why this is needed.

File: atomic_reactor/plugins/pre_inject_parent_image.py [hunk 5]
```diff
@@ -114,8 +115,19 @@ def select_new_parent_image(self):
...
+        new_parent_image = ImageName.parse(self._new_parent_image)
+
+        if new_parent_image.registry != self.workflow.builder.base_image.registry:
+            new_parent_image.registry = self.workflow.builder.base_image.registry
+
+        if organization:
+            self.workflow.builder.base_image.enclose(organization)
+            new_parent_image.enclose(organization)
+
+        self._new_parent_image = new_parent_image.to_str()
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/pre_inject_parent_image.py]: ok

File: atomic_reactor/plugins/pre_pull_base_image.py (hunk 4 above)
Comment (rcerven) [reply atomic_reactor/plugins/pre_pull_base_image.py]: ok

File: atomic_reactor/plugins/pre_inject_parent_image.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/pre_inject_parent_image.py]: ok

File: atomic_reactor/plugins/pre_change_from_in_df.py [hunk 6]
```diff
@@ -58,14 +62,22 @@ def run(self):
...
             raise ParentImageUnresolved("Parent image(s) unresolved: {}".format(unresolved))
 
-        missing = [df_img for df_img in dfp.parent_images
+        # enclose images from dfp
+        enclosed_parent_images = []
+        for df_img in dfp.parent_images:
+            parent = ImageName.parse(df_img)
+            if organization:
+                parent.enclose(organization)
+            enclosed_parent_images.append(parent.to_str())
```
[the remark this answers is not in the document]
Comment (rcerven) [reply atomic_reactor/plugins/pre_change_from_in_df.py]: ok

File: atomic_reactor/plugins/pre_inject_parent_image.py [hunk 7]
```diff
@@ -69,6 +70,7 @@ def run(self):
         self.find_repositories()
         self.select_new_parent_image()
         self.validate_new_parent_image()
+        self.adjust_new_parent_image()
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_inject_parent_image.py]: Don't we want to adjust before we validate it?

Comment (rcerven) [reply atomic_reactor/plugins/pre_inject_parent_image.py]: yeah we do :) moved

**Comment by lcarva** [comment]
Updated release notes draft:
> - `registries_organization` now used for source registry
